test/index.test.ts

```diff
@@ -771,6 +771,48 @@ describe('Unit | Utility | changeset', () => {
     expect(dummyModel.name.short).toBe('foo');
   });
 
+  it('#set nested objects can contain arrays', () => {
+    expect.assertions(5);
+
+    dummyModel.contact = {
+      emails: ['bob@email.com', 'the_bob@email.com']
+    };
+
+    expect(get(dummyModel, 'contact.emails')).toEqual(['bob@email.com', 'the_bob@email.com']);
+
+    const dummyChangeset = Changeset(dummyModel, lookupValidator(dummyValidations));
+
+    expect(dummyChangeset.get('contact.emails')).toEqual(['bob@email.com', 'the_bob@email.com']);
+
+    dummyChangeset.set('contact.emails.0', 'fred@email.com');
+
+    expect(dummyChangeset.get('contact.emails.0')).toEqual('fred@email.com');
+
+    dummyChangeset.rollback();
+    expect(dummyChangeset.get('contact.emails')).toEqual(['bob@email.com', 'the_bob@email.com']);
+
+    dummyChangeset.set('contact.emails.0', 'fred@email.com');
+    dummyChangeset.set('contact.emails.1', 'the_fred@email.com');
+
+    // This is still in object format
+    // expect(dummyChangeset.get('contact.emails')).toEqual(['fred@email.com', 'the_fred@email.com']);
+
+    dummyChangeset.execute();
+    expect(dummyModel.contact.emails).toEqual(['fred@email.com', 'the_fred@email.com']);
+  });
```

Comment:
Could we also have some assertions for...

- `changeset.changes`
- `changeset.isValid`
- `changeset.save`

Reply:
added